app/helpers/application_helper/button/host_introspect_provide.rb

```diff
@@ -0,0 +1,17 @@
+class ApplicationHelper::Button::HostIntrospectProvide < ApplicationHelper::Button::Basic
+  needs :@record
+
+  def visible?
+    proper_record? && host_manageable?
+  end
+
+  private
+
+  def proper_record?
+    @record.class == ManageIQ::Providers::Openstack::InfraManager::Host
```

Comment:
Yes, we want to avoid that -- testing for particular class -- and use the "supports feature" toolset. 

But the goal of this PR (and the effort) is to split the code for the individual buttons. So if the feature is missing on the underlying model, then you can create an issue and get that fixed in a follow up PR.

Seeing "Openstack" in the name, you might want to ask @tzumainn to assign someone to add the missing feature. Or add you yourself if it's trivial enough.